src/Collection.php

```diff
@@ -89,7 +89,7 @@ public function averages(): CollectionInterface
 
     public function cache(?CacheItemPoolInterface $cache = null): CollectionInterface
     {
-        return new self((new Operation\Cache())()($cache ?? new ArrayAdapter()), [$this]);
+        return new self((new Operation\Cache())()($cache ?? new ArrayAdapter()), [$this->squash()]);
```

Comment:
to make sure I got this right: after this change calling `cache` will always eagerly evaluate all the previous operations?

If that's the case, I'm not sure this is really a "fix" but a change in functionality. It's fine to introduce it but I think it might be good to have it based on a boolean parameter and not enabled by default so as to not change the previous behviour. WDYT?

Reply:
That's the whole point of this operation. Using `cache` should do what it says.

In the [following example](https://github.com/loophp/collection/issues/264#issuecomment-1808384744), the `cache` operation doesn't do at all what it is supposed to do, this PR fix this.

And you're totally right when you say that this is eagerly evaluating the Collection. I don't see we could do that differently at the moment. I wish there would be a better option to avoid evaluating eagerly, but I couldn't find anything better yet.

Reply:
I think an important distinction might be needed here, but correct me if I'm wrong: in that example, `cache + all` is not doing what the user expects. Is this is a general issue with `cache`, is it working in any other cases?

If it is working in other cases then the behaviour change in this PR is breaking BC, which is why I was suggesting to only add `squash` based on a param, eg:

```php
public function cache(?CacheItemPoolInterface $cache = null, boolean $shouldCache = false): CollectionInterface
    {
        $withSquash = $shouldCache ? [$this->squash()] : [];
        return new self((new Operation\Cache())()($cache ?? new ArrayAdapter()), $withSquash);
    }
```

Reply:
I'll check when I'm back in front of my laptop, away from computer right now :S

Reply:
Hey @drupol, I assume you haven't had the chance to look at this yet? Please feel free to reopen when you do :)